flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManager.java

```diff
@@ -31,7 +32,7 @@
  * FileMergingSnapshotManager provides an interface to manage files and meta information for
  * checkpoint files with merging checkpoint files enabled. It manages the files for ONE single task
  * in TM, including all subtasks of this single task that running in this TM. There is one
- * FileMergingSnapshotManager for each task per task manager.
+ * FileMergingSnapshotManager for each job per task manager.
```

Comment:
If a job has multiple tasks on the same TM (not sub-tasks), do they share the same FileMergingSnapshotManager?
I mean something like "(Source 0/1)"" (MAP 0/1)", but source and map belong to the same job

Reply:
> If a job has multiple tasks on the same TM (not sub-tasks), do they share the same FileMergingSnapshotManager?

Yes, they share the same `FileMergingSnapshotManager`, and "(Source 0/1)" " (MAP 0/1)" have different `OperatorId`, so they would have their own subdirectories.